ADAL/src/request/ADAuthenticationRequest+AcquireToken.m

```diff
@@ -500,4 +506,19 @@ - (void)requestTokenByCode:(NSString *)code
               completion:completionBlock];
 }
 
+- (void)tryRefreshToken:(ADAuthenticationCallback)completionBlock
+{
+    [[ADTelemetry sharedInstance] startEvent:[self telemetryRequestId] eventName:AD_TELEMETRY_EVENT_ACQUIRE_TOKEN_SILENT];
+    ADAcquireTokenSilentHandler* request = [ADAcquireTokenSilentHandler requestWithParams:_requestParams];
+    [request acquireTokenByRefreshToken:_refreshToken
+                              cacheItem:[ADTokenCacheItem new]
```

Comment:
do we need to pass a dummy cache item? Can we just pass nil?

Reply:
Seems the function is designed to not updating cache if that value is nil :)

Reply:
I see it creates a new item itself if there's none passed in the acquireTokenByRefreshToken:cacheItem:completionBlock:
ADTokenCacheItem* resultItem = (cacheItem) ? cacheItem : [ADTokenCacheItem new];

Reply:
It is this line: https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/release/2.6.0/ADAL/src/request/ADAcquireTokenSilentHandler.m#L151

But I realize that such a function will delete tokens if server rejects the RT, which is not what we want. I will figure out a way to avoid it tmr :)

Reply:
@oldalton After second look, I find that the code should be fine because of this line:
https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/release/2.6.0/ADAL/src/cache/ADTokenCacheAccessor.m#L402

In other words, If the RT provided by developer is rejected and there is another RT in cache with the same cache key, it won't delete the RT in cache unless they are EXACTLY the same RT. What do you think?

I have added one more uint test case for this scenario.

Reply:
Yes, that should work